src/hb-serialize.hh

```diff
@@ -415,20 +432,20 @@ struct hb_serialize_context_t
   Type *start_embed (const Type &obj) const
   { return start_embed (hb_addressof (obj)); }
 
-  /* Following two functions exist to allow setting breakpoint on. */
-  void err_ran_out_of_room () { this->ran_out_of_room = true; }
-  void err_offset_overflow () { this->offset_overflow = true; }
-  void err_other_error () { this->successful = false; }
+  void set_error (hb_serialize_error_type_t err_type)
+  {
+    errors = errors | err_type;
+  }
 
   template <typename Type>
   Type *allocate_size (unsigned int size)
   {
-    if (unlikely (!this->successful)) return nullptr;
+    if (unlikely (in_error ())) return nullptr;
 
     if (this->tail - this->head < ptrdiff_t (size))
     {
-      err_ran_out_of_room ();
-      this->successful = false;
+      set_error (HB_SERIALIZE_ERR_OUT_OF_ROOM);
+      set_error (HB_SERIALIZE_ERR_OTHER);
```

Comment:
This code just needs to set out-of-room. No need to set OTHER based on suggestions above.